### PR TITLE
Add new encryption_property that simplifies setup

### PR DESCRIPTION
**Pull request by jamie-chang-globality**

### CODE REVIEW

File: microcosm_postgres/encryption/v2/encoders.py [hunk 1]
```diff
@@ -24,6 +30,8 @@ def decode(self, value: str) -> T:
 
 
 class StringEncoder(Encoder[str]):
+    sa_type = sqlalchemy.String
```
Comment (jamie-chang-globality) [inline microcosm_postgres/encryption/v2/encoders.py:33]: Encoders now contain the default type that the column expects to have

File: microcosm_postgres/encryption/v2/column.py [hunk 2]
```diff
@@ -10,39 +17,107 @@
...
 
-    Queries on the encryption field will only be performed on the unencrypted rows.
-    """
-    encrypted_field = f"{key}_encrypted"
-    unencrypted_field = f"{key}_unencrypted"
 
-    @hybrid_property
-    def _prop(self) -> T:
-        encrypted = getattr(self, encrypted_field)
+class encryption(hybrid_property[T], Generic[T]):
```
[the remark this answers is not in the document]
Comment (jamie-chang-globality) [reply microcosm_postgres/encryption/v2/column.py:23]: backwards compatible with the old `encryption` and also it's the same as `hybrid_properties`

File: microcosm_postgres/encryption/v2/column.py [hunk 3]
```diff
@@ -10,39 +17,107 @@
...
+    @overload
+    def __init__(
+        self,
+        key: str,
+        encryptor: Encryptor,
+        encoder: Encoder[T],
+        *,
+        column_type: Any = NOT_SET,
```
Comment (JP-Globality) [inline microcosm_postgres/encryption/v2/column.py:31]: type as `object`?

Comment (JP-Globality) [reply microcosm_postgres/encryption/v2/column.py:31]: Ignore

**Review by JP-Globality — APPROVED** [review]

**Review by dumpstate — APPROVED** [review]

**Pull request merged**